### PR TITLE
Add SiliconDust's HdHomerun standalone DVR engine Rockon

### LINKED ISSUE #278: Add HDHomeRun DVR engine Rockon
HDHomeRun Tuners can record (with a valid subscription) live tv using a separate DVR engine that can be installed under windows or Linux. This makes a good Rockstor Rockon to add broadcast TV recordings to one's media library.

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #278 .

General information on project
This pull request proposes to add a new rock-on for the following project:

name: HDHomeRun DVR engine from SiliconDust
website: https://www.silicondust.com
description: standalone DVR engine that work in conjunction with SiliconDust's HDHomeRun series of tuners that don't have the DVR built-in. Requires a valid subscription and recent HDHomeRun hardware
Information on docker image
docker image: https://hub.docker.com/repository/docker/jackdock96/hdh_dvr
is an official docker image available for this project?: Not at this time. I created this docker image to self-update with the latest engine and also offer multi-arch images to support Rockstor's direction of installing on ARM architecture systems, so the docker container doesn't have to be maintained in the repository too frequently.
Note: because the dvr engine uses multi-cast the Rockon is currently configured to use the `net=host` option. Unless one implements a `macvlan` or uses tools like `pipework`, multi-cast from LAN to container and vice-versa is not possible. I assume, Rockstor's Rocknets might be able to achieve that, but until we are completely cut over to 4.x on OpenSuse, this is the only option.
Plan is to switch to an official image if/when it becomes available

### Checklist
- [x] Passes [JSONlint](https://jsonlint.com) validation
- [x] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [x] `"description"` object lists and links to the docker image used
- [x] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [x] `"website"` object links to project's main website

### CODE REVIEW

**Comment by Hooverdan96** [comment]
The amd64 version works very well on my environments. I have not been able to test the ARM architecture container versions on "real" hardware at this time, though.

File: root.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "Gollum": "gollum.json",
     "HAProxy-Letsencrypt": "haproxy-letsencrypt.json",
     "HandBrake": "handbrake.json",
-    "Headphones": "headphones.json",
+    "HDHomeRun DVR":"hdhomerun_dvr.json",
+	"Headphones": "headphones.json",
```
Comment (phillxnet) [inline root.json:26]: @Hooverdan96  This looks to have acquired a 'stray' tab char!
```suggestion
    "Headphones": "headphones.json",
```
The file is probably out of kilter with the more common 2 space json in this repo but changing that now would likely upset some merges now. But no tabs if we can help it.

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for your continued effort in this repo by the way. Much appreciated.

**Comment by Hooverdan96** [comment]
Thanks @phillxnet, accepted your changes ... always good to have more eyes on this :).

**Comment by phillxnet** [comment]
Just checked and this image was updated 24 days ago:
https://hub.docker.com/search?q=jackdock96%2Fhdh_dvr&type=image

But we have another host network requirement. But it's not quite a wild as say deluge that I recently closed an issue for to extent it to host networking:
https://github.com/rockstor/rockon-registry/issues/205

@FroggyFlox Your call on this one. I'm game if you are.

**Comment by Hooverdan96** [comment]
@phillxnet, it was updated since I did the updating. I am maintaining a docker image (updating itself with the latest version of the executable whenever restarted). Fortunately, the docker container updates currently only involve building it with a new version of alpine. If/when an official docker is released (not sure if that ever happens), we can obviously go with that one.
and, yes, the net=host is again required for the upnp stuff, otherwise it cannot find the tuner. I've tried. I hope, there will be some type of "multicast" solution that easily integrates with docker in the future...

**Comment by Hooverdan96** [comment]
I completely messed up my fork and ended up pushing not the right stuff. I am going to close this and replace it with #301 

My apologies.

**Pull request closed without merging**